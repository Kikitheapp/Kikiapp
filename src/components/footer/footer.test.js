import React from 'react';
import {render, screen} from '@testing-library/react';

import renderer  from 'react-test-renderer';

import '@testing-library/jest-dom';
import Footer from './footer';

describe("Footer", () => {

  const year = new Date().getFullYear();

  const footerText1 = "Kiki for the Future®";
  const footerText2 = "Follow us";
  const footerText3 = `© 2021-${year} Kiki for the Future`;

  it("renders", async () => {
      

    // render the Footer component
    render(<Footer />);

    // check that the footer text is correct
    expect(screen.getByText(footerText1)).toBeInTheDocument();
    expect(screen.getByText(footerText2)).toBeInTheDocument();
    expect(screen.getByText(footerText3)).toBeInTheDocument();

  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
 });

