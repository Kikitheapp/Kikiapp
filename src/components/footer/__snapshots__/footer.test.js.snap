// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer
  className="container-fluid bg-primary"
>
  <h5>
    Kiki for the Future®
  </h5>
  <h5>
    Follow us
  </h5>
  <div>
    <a
      className="social-icons"
      href="https://www.facebook.com/kikitheapp/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
        />
      </svg>
    </a>
    <a
      className="social-icons"
      href="https://www.instagram.com/kikitheapp/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224,202.66A53.34,53.34,0,1,0,277.36,256,53.38,53.38,0,0,0,224,202.66Zm124.71-41a54,54,0,0,0-30.41-30.41c-21-8.29-71-6.43-94.3-6.43s-73.25-1.93-94.31,6.43a54,54,0,0,0-30.41,30.41c-8.28,21-6.43,71.05-6.43,94.33S91,329.26,99.32,350.33a54,54,0,0,0,30.41,30.41c21,8.29,71,6.43,94.31,6.43s73.24,1.93,94.3-6.43a54,54,0,0,0,30.41-30.41c8.35-21,6.43-71.05,6.43-94.33S357.1,182.74,348.75,161.67ZM224,338a82,82,0,1,1,82-82A81.9,81.9,0,0,1,224,338Zm85.38-148.3a19.14,19.14,0,1,1,19.13-19.14A19.1,19.1,0,0,1,309.42,189.74ZM400,32H48A48,48,0,0,0,0,80V432a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V80A48,48,0,0,0,400,32ZM382.88,322c-1.29,25.63-7.14,48.34-25.85,67s-41.4,24.63-67,25.85c-26.41,1.49-105.59,1.49-132,0-25.63-1.29-48.26-7.15-67-25.85s-24.63-41.42-25.85-67c-1.49-26.42-1.49-105.61,0-132,1.29-25.63,7.07-48.34,25.85-67s41.47-24.56,67-25.78c26.41-1.49,105.59-1.49,132,0,25.63,1.29,48.33,7.15,67,25.85s24.63,41.42,25.85,67.05C384.37,216.44,384.37,295.56,382.88,322Z"
        />
      </svg>
    </a>
    <a
      className="social-icons"
      href="https://www.patreon.com/kikitheapp"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 194.8c0 101.3-82.4 183.8-183.8 183.8-101.7 0-184.4-82.4-184.4-183.8 0-101.6 82.7-184.3 184.4-184.3C429.6 10.5 512 93.2 512 194.8zM0 501.5h90v-491H0v491z"
        />
      </svg>
    </a>
    <a
      className="social-icons"
      href="https://www.tiktok.com/@kikitheapp"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448,209.91a210.06,210.06,0,0,1-122.77-39.25V349.38A162.55,162.55,0,1,1,185,188.31V278.2a74.62,74.62,0,1,0,52.23,71.18V0l88,0a121.18,121.18,0,0,0,1.86,22.17h0A122.18,122.18,0,0,0,381,102.39a121.43,121.43,0,0,0,67,20.14Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="pt-2"
  >
    <a
      className="text-charcoal text-decoration-none"
      href="/privacypolicy"
    >
      Privacy Policy
    </a>
     - 
    <a
      className="text-charcoal text-decoration-none"
      href="/termsofuse"
    >
      Terms of Use
    </a>
  </div>
  <div
    className="pt-2"
  >
    <small>
      © 2021-
      2024
       Kiki for the Future
      <sup>
        ®
      </sup>
    </small>
  </div>
</footer>
`;
